js/algoliasearch/internals/adminhtml/admin_scripts.js

```diff
@@ -75,4 +75,33 @@ algoliaAdminBundle.$(function($) {
 			$synonymsRows.hide();
 		}
 	}
+	
```

Comment:
It shows the text input field for custom attribute only when you have selected `[use custom attribute]` option in select box.